internal/js/modules/k6/browser/common/frame.go

```diff
@@ -1473,6 +1473,8 @@ func (f *Frame) selectOption(selector string, values []any, opts *FrameSelectOpt
 func (f *Frame) SetContent(html string, _ *FrameSetContentOptions) error {
 	f.log.Debugf("Frame:SetContent", "fid:%s furl:%q", f.ID(), f.URL())
 
+	// TODO(@inancgumus): Respect the FrameSetContentOptions before executing the action.
```

Comment:
Out of curiosity, would it be enough by wrapping the context with the timeout or do we need something else? It would be good to have a suggestion regarding how to fix it so whoever will look into it then they might have a better a starting point.

Reply:
Yes, a suggestion would be good. But it's complicated to explain since the option not only involves a timeout, it also involves waiting for a "lifecycle" event. I've added a small suggestion: [41af72a](https://github.com/grafana/k6/pull/4538/commits/41af72a196c7141b4aced1acb7a8f39b0f81ef7a)